Covid19Radar/Covid19Radar/Repository/ExposureConfigurationRepository.cs

```diff
@@ -0,0 +1,340 @@
+        {
+            _loggerService.StartMethod();
+
+            ExposureConfiguration currentExposureConfiguration = null;
+
+            if (File.Exists(_currentExposureConfigurationPath))
+            {
+                _loggerService.Debug("ExposureConfiguration file is found.");
+
+                try
+                {
+                    string exposureConfigurationAsJson = await LoadAsync(_currentExposureConfigurationPath);
+
+                    currentExposureConfiguration = JsonConvert.DeserializeObject<ExposureConfiguration>(exposureConfigurationAsJson);
+
+                    if (!IsDownloadedExposureConfigurationOutdated(AppConstants.ExposureConfigurationFileDownloadCacheRetentionDays))
+                    {
+                        return currentExposureConfiguration;
+                    }
+                    else
+                    {
+                        _loggerService.Info($"ExposureConfiguration is found but the file is outdated.");
+                    }
+                }
+                catch (IOException exception)
+                {
+                    _loggerService.Exception("IOException. ExposureConfiguration file has been deleted.", exception);
+                    RemoveExposureConfigurationInternal();
+                }
+                catch (JsonException exception)
+                {
+                    _loggerService.Exception("JsonException. ExposureConfiguration file has been deleted.", exception);
+                    RemoveExposureConfigurationInternal();
+                }
+            }
+
+            if (currentExposureConfiguration is null)
+            {
+                currentExposureConfiguration = new ExposureConfiguration();
+            }
+
+            await _serverConfigurationRepository.LoadAsync();
+            string url = _serverConfigurationRepository.ExposureConfigurationUrl;
+
+            ExposureConfiguration newExposureConfiguration = null;
+
+            var response = await _client.GetAsync(url);
+            if (response.IsSuccessStatusCode)
+            {
+                string exposureConfigurationAsJson = await response.Content.ReadAsStringAsync();
+                _loggerService.Debug(exposureConfigurationAsJson);
+
+                try
+                {
+                    newExposureConfiguration = JsonConvert.DeserializeObject<ExposureConfiguration>(exposureConfigurationAsJson);
+                    SetExposureConfigurationDownloadedDateTime(_dateTimeUtility.UtcNow);
+                }
+                catch (JsonException exception)
+                {
+                    _loggerService.Exception("JsonException.", exception);
+                }
+
+            }
+            else
+            {
+                _loggerService.Warning($"Download ExposureConfiguration failed from {url}");
+            }
+
+            if (newExposureConfiguration is null)
+            {
+                return currentExposureConfiguration;
```

Comment:
`_loggerService.EndMethod();` を呼び出してからリターンすべきです。

```suggestion
                _loggerService.EndMethod();
                return currentExposureConfiguration;
```

Reply:
6a3cbf0 で対応しました！